keps/sig-instrumentation/4785-crdmetrics-controller/README.md

```diff
@@ -0,0 +1,1061 @@
+* allows for Custom Resource metrics generation based on their schemata,
+* while providing cluster-scoped managed resources (`CRDMetricsResource`) that 
+  allows defining the collection configuration for generating metrics 
+  on-the-fly, and,
+* while being able to accommodate for multiple configuration parsing techniques
+  and expression non-turing languages,
+* all while conforming to, and improving the existing Custom Resource State API 
+  offered by Kube State Metrics without hindering the maintainability and the
+  scalability of the controller.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+The KEP does **not** target a controller which:
+* overlaps with Kube State Metrics' goals in any way except for the Custom 
+  Resource State API, or,
+* offers any stability guarantees for the metrics generated using its managed 
+  resource(s).
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The proposal targets the incubation and incorporation of a controller capable
+of essentially doing all that Kube State Metrics' Custom Resource State API
+offers, in additional to various benefits of its own owing to the controller
+lifecycle it is based upon.
+
+The controller aims to replace the existing Kube State Metrics' Custom Resource 
+State feature-set and be significantly much more maintainable and scalable than
+its predecessor while enabling the community the freedom to extend the supported
+set of DSLs to parse the configuration in a language that they are familiar
+with, instead of forcing them up a steep learning-curve for a self-defined DSL
+that they have to live with, as is the case with Kube State Metrics' Custom
+Resource State feature-set.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a cluster admin, I want to be able to express my Kube State Metrics' Custom 
+Resource State configurations in dedicated CRs, so I can benefit from everything
+the controller lifecycle has to offer, for e.g., not having to mount volumes and 
+redeploy their modifications, event emissions on metrics generation, 
+multiple managed resources to isolate configuration logic, etc.
+
+#### Story 2
+
+As a cluster admin, I want to be able to express my Kube State Metrics' Custom
+Resource State configurations in a non-turing language that is well-known
```

Comment:
What's non-Turing language? Do you mean “Turing-incomplete” perhaps?

Reply:
Hmm, right. Not sure why they say "non-Turing complete" instead of just "Turing incomplete" [here](https://github.com/google/cel-go/blob/f8ecaa2a9a5ec4e9585e92794e66bef7cceb41b8/README.md?plain=1#L6-L8). I'll fix this.